src/core/public/chrome/nav_links/nav_links_service.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { sortBy } from 'lodash';
+import { BehaviorSubject, ReplaySubject } from 'rxjs';
+import { map, takeUntil } from 'rxjs/operators';
+import { NavLinkWrapper, NavLinkUpdateableFields } from './nav_link';
+import { ApplicationStart } from '../../application';
+
+interface StartDeps {
+  application: ApplicationStart;
+}
+
+export class NavLinksService {
+  private readonly stop$ = new ReplaySubject(1);
+
+  public start({ application }: StartDeps) {
+    const navLinks$ = new BehaviorSubject<ReadonlyArray<NavLinkWrapper>>(
+      application.availableApps.map(
+        app =>
+          new NavLinkWrapper({
+            ...app,
+            // Either rootRoute or appUrl must be defined.
+            appUrl: (app.rootRoute || app.appUrl)!,
+          })
+      )
+    );
+    const forceAppSwitcherNavigation$ = new BehaviorSubject(false);
+
+    return {
+      /**
+       * Get an observable for a sorted list of navlinks.
+       */
+      getNavLinks$: () => {
+        return navLinks$.pipe(
+          map(sortNavLinks),
+          takeUntil(this.stop$)
+        );
+      },
+
+      /**
+       * Get the state of a navlink at this point in time.
+       * @param id
+       */
+      get(id: string) {
+        const link = navLinks$.value.find(l => l.id === id);
+        return link ? link.properties : undefined;
```

Comment:
If you want to avoid the explicit return of undefined and the ternary, you could write as:

```js
return link && link.properties;
```